src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -3564,6 +3573,42 @@ instruct vmask_reinterpret_diff_esize(vRegMask dst, vRegMask_V0 src, vReg tmp) %
   ins_pipe(pipe_slow);
 %}
 
+// ------------------------------ Vector shuffle -------------------------------
+
+instruct loadshuffleB(vReg dst) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_BYTE);
+  match(Set dst (VectorLoadShuffle dst));
+  effect(TEMP_DEF dst);
+  format %{ "loadshuffleB $dst, $dst" %}
+  ins_encode %{
+    // For T_BYTE, no need to do anything
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct loadshuffleX(vReg dst, vReg src) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_DOUBLE ||
+            Matcher::vector_element_basic_type(n) == T_LONG ||
+            Matcher::vector_element_basic_type(n) == T_FLOAT ||
+            Matcher::vector_element_basic_type(n) == T_INT ||
+            Matcher::vector_element_basic_type(n) == T_SHORT);
+  match(Set dst (VectorLoadShuffle src));
+  effect(TEMP_DEF dst);
+  format %{ "loadshuffleX $dst, $src" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    __ vsetvli_helper(bt, Matcher::vector_length(this));
+    if (bt == T_SHORT) {
+      __ vsext_vf2(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
```

Comment:
I prefer `vzext_vf2/4/8` which does zero extension for the source indexes.

Reply:
done.